GVFS/GVFS.NativeHooks.Common/common.posix.cpp

```diff
@@ -31,22 +31,23 @@ PATH_STRING GetGVFSPipeName(const char *appName)
     }
     
     PATH_STRING finalRootPath(GetFinalPathName(enlistmentRoot));
-    size_t resultLength = strlcpy(enlistmentRoot, finalRootPath.c_str(), sizeof(enlistmentRoot));
-    if (resultLength >= sizeof(enlistmentRoot))
+    size_t enlistmentRootLength = finalRootPath.length();
+    // allow an extra byte in case we need to add a trailing slash
+    if (enlistmentRootLength + 2 > sizeof(enlistmentRoot))
     {
         die(ReturnCode::PipeConnectError,
-            "Could not copy finalRootPath: %ls, insufficient buffer. resultLength: %d, sizeof(enlistmentRoot): %d\n",
+            "Could not copy finalRootPath: %ls, insufficient buffer. enlistmentRootLength: %d, sizeof(enlistmentRoot): %d\n",
             finalRootPath.c_str(),
-            resultLength,
+            enlistmentRootLength,
             sizeof(enlistmentRoot));
     }
     
-    size_t enlistmentRootLength = strlen(enlistmentRoot);
-    if ('/' != enlistmentRoot[enlistmentRootLength - 1])
+    memcpy(enlistmentRoot, finalRootPath.c_str(), enlistmentRootLength);
+    if (enlistmentRootLength == 0 || enlistmentRoot[enlistmentRootLength - 1] != '/')
```

Comment:
On self-review, I think the [existing code](https://github.com/Microsoft/VFSForGit/blob/master/GVFS/GVFS.NativeHooks.Common/common.mac.cpp#L45) and my first try at this both could fail if `enlistmentRootLength` is zero as they test `enlistmentRoot[enlistmentRootLength - 1]`.  It's possible the path can't be empty, but I figure it doesn't hurt to be careful here.